guides/source/caching_with_rails.md

```diff
@@ -30,117 +33,115 @@ config.action_controller.perform_caching = true
+When your application recieves its first request to this page Rails will write
+a new cache to a unique key. A key looks something like this:
 
-```html+erb
-<% cache(action: 'recent', action_suffix: 'all_products') do %>
-  All available products:
+```html
+views/products/1-201505056193031061005000/bea67108094918eeba42cd4a6e786901
 ```
 
-and you can expire it using the `expire_fragment` method, like so:
+The number in the middle is the `product_id` followed by the timestamp value in
+the `updated_at` cell of the product table. Rails uses the timestamp value to
+make sure it is not serving stale data. If the value of `updated_at` has
+changed a new key will be generated, Rails will write a new cache to that key,
+and the old cache written to the old key will never be used again. This is
+called key-based expiration.
 
-```ruby
-expire_fragment(controller: 'products', action: 'recent', action_suffix: 'all_products')
-```
+Cache fragments will also be expired when the view fragment changes (e.g., the
+HTML in the view changes). The string of characters at the end of the key is a
+template tree digest. It is an md5 hash computed based on the contents of the
+view fragment you are caching. If you change the view fragment, the md5 hash
+will change, expiring the existing file.
+
+TIP: Cache stores like Memcached will automatically delete old cache files.
 
-If you don't want the cache block to bind to the action that called it, you can also use globally keyed fragments by calling the `cache` method with a key:
+If you want to cache a fragment under certain conditions, you can use
+`cache_if` or `cache_unless`
 
 ```erb
-<% cache('all_available_products') do %>
-  All available products:
+<% cache_if admin?, product do %>
+  <%= render product %>
 <% end %>
 ```
 
-This fragment is then available to all actions in the `ProductsController` using the key and can be expired the same way:
+### Russian Doll Caching
 
-```ruby
-expire_fragment('all_available_products')
-```
-If you want to avoid expiring the fragment manually, whenever an action updates a product, you can define a helper method:
-
-```ruby
-module ProductsHelper
-  def cache_key_for_products
-    count          = Product.count
-    max_updated_at = Product.maximum(:updated_at).try(:utc).try(:to_s, :number)
-    "products/all-#{count}-#{max_updated_at}"
-  end
-end
-```
+You may want to nest cached fragments inside other cached fragments. This is
+called Russian doll caching.
 
-This method generates a cache key that depends on all products and can be used in the view:
+The advantage of Russian doll caching is that if a single product is updated, all the other inner fragments can be reused when regenerating the outer fragment.
 
-```erb
-<% cache(cache_key_for_products) do %>
-  All available products:
-<% end %>
-```
+As explained in the previous section, a cached file will expire if the value of
+`updated_at` changes in a record the cached file directly depends on. However,
```

Comment:
i found this a little hard to scan. what do you think about '[...] changes in a record on which the cached file directly depends'?